charon.py

```diff
@@ -221,4 +185,44 @@ async def on_member_join(member):
         f'Feel free to dial \"{COMMAND_PREFIX}help\" if you require any assistance.\n'
         f'...and as always, it is a pleasure having you with us again, {member.name}.')
 
-bot.run(TOKEN)
+@bot.event
+async def on_reaction_add(reaction, user):
+    global parties
+    for party in parties:
+        if user.name == bot.user.name:
+            continue
+        if party.isMatchJoinEmoji(reaction):
+            party.addMember(user.name)
+            await reaction.message.edit(embed = party.getEmbed())
+            break
+        if party.isMatchCloseEmoji(reaction, user):
+            party.close()
+            await reaction.message.edit(embed = party.getEmbed())
+            parties.remove(party)
+            break
+
+@bot.event
+async def on_reaction_remove(reaction, user):
+    for party in parties:
+        if party.isMatchJoinEmoji(reaction):
+            party.removeMember(user.name)
+            await reaction.message.edit(embed = party.getEmbed())
+            break
+
+# seconds of running background loop
+BACKGROUND_LOOP_TIME = 60
+# This function checks if parties are inactive
+# and cleans up them up if they are
+async def update_parties():
```

Comment:
There is a decorator that might be available here
https://discordpy.readthedocs.io/en/latest/ext/tasks/index.html?highlight=task#discord.ext.tasks.loop

Reply:
will check it out